cmd/outline-ss-server/metrics.go

```diff
@@ -149,143 +271,112 @@ func (c *tunnelTimeCollector) stopConnection(ipKey IPKey) {
 	}
 }
 
-func newTunnelTimeCollector(ip2info ipinfo.IPInfoMap, registerer prometheus.Registerer) *tunnelTimeCollector {
-	return &tunnelTimeCollector{
-		ip2info:       ip2info,
-		activeClients: make(map[IPKey]*activeClient),
+type outlineMetricsCollector struct {
+	ipinfo.IPInfoMap
```

Comment:
Let's not embed these fields. It makes it harder to understand, and you are not leveraging it anyway. You have to reimplement the methods you are exposing. Making them variables will clarify what the type really exposes.

Reply:
I see the IPInfoMap still embedded.

Reply:
Ah that's how it is today because of the `TCPMetrics` interface. It's a bit more tricky to remove and is better suited to #200 where I decouple that. Let me make sure we stop embedding there as well, but will leave this as-is here.

Reply:
Just confirmed I indeed removed the embedded `IPInfoMap` in https://github.com/Jigsaw-Code/outline-ss-server/pull/200/files#diff-b632867f1b855f9924106632a21b7093d1d0494e19eb55427589e425c9ff468bL281